vyper/codegen/expr.py

```diff
@@ -189,12 +189,10 @@ def __init__(self, node, context):
         self.ir_node.source_pos = getpos(self.expr)
 
     def parse_Int(self):
-        # Literal (mostly likely) becomes int256
-        if self.expr.n < 0:
-            return IRnode.from_list(self.expr.n, typ=BaseType("int256", is_literal=True))
-        # Literal is large enough (mostly likely) becomes uint256.
-        else:
-            return IRnode.from_list(self.expr.n, typ=BaseType("uint256", is_literal=True))
+        typ_ = self.expr._metadata.get("type")
+        if not typ_:
+            raise CompilerPanic("Type of integer literal is unknown")
+        return IRnode.from_list(self.expr.n, typ=BaseType(str(typ_), is_literal=True))
```

Comment:
use `new_type_to_old_type` here

Reply:
`new_type_to_old_type` does not allow the `is_literal` property to be set. Should I add `is_literal` as a kwarg to `new_type_to_old_type`?